### PR TITLE
[receiver/jaegerreceiver] Thrift packets processing metrics

### PR DESCRIPTION
**Pull request by larsn777**
**Description:** <Describe what has changed.>
Added metrics for the number of Thrift packets processed and rejected.
These metrics will help to monitor possible data loss on Jaeger receiver component.

**Link to tracking Issue:** 
#34462

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/jaegerreceiver/metrics.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package jaegerreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jaegerreceiver"
+
+import (
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+)
+
+const (
+	DroppedPacketsStat   = "thrift.udp.server.packets.dropped"
```
Comment (yurishkuro) [inline receiver/jaegerreceiver/metrics.go:11]: To clarify, these are the names that thrift server is using to report the metrics, but you're re-mapping them to counters generated from metadata? I would suggest adding a comment to that effect.

Any reason not to keep the metric names identical to Jaeger, for better compatibility?

Comment (larsn777) [reply receiver/jaegerreceiver/metrics.go:11]: Hello 

> To clarify, these are the names that thrift server is using to report the metrics, but you're re-mapping them to counters generated from metadata?

Yes, that's right.

> Any reason not to keep the metric names identical to Jaeger, for better compatibility?

Unfortunately, it is currently not possible to retrieve counter names from the Jaeger SDK as t[hese names are specified using private/non-exportable field tags](https://github.com/jaegertracing/jaeger/blob/3bc0d35849ac78fe888b33f4a353bc6ee78efa78/cmd/agent/app/servers/tbuffered_server.go#L52).

> I would suggest adding a comment to that effect.

I can add a corresponding comment to the counters names definition.

Comment (yurishkuro) [reply receiver/jaegerreceiver/metrics.go:11]: > Unfortunately, it is currently not possible to retrieve counter names from the Jaeger SDK

You can copy the names but keep them the same. Right now they are changed to completely different metrics.

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
@larsn777 looks like there are merge conflicts that need to be resolved

**Review by mwear — APPROVED** [review]
LGTM. @larsn777, please resolve the merge conflicts when you get a chance.

**Comment by larsn777** [comment]
@yurishkuro @mwear  Sorry for the late response.
I resolved the conflicts and also committed a [small fix](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34464/commits/729aff084e1a5ce79dae27a06e9b14271ef73abc) and [update for the autogenerated telemetry code](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34464/commits/716a99e5d3f4dd9ab78df78bc2e729a3e7ed6a8f).

**Pull request closed without merging**